src/app/edit.component/edit.component.ts

```diff
@@ -1,54 +1,44 @@
-import { Component, OnInit } from '@angular/core'
+import { Component, OnInit, Input } from '@angular/core'
 import { Todo } from '../todo.data/todo';
 import { HeroService } from '../services/todo.service';
 import {Router} from '@angular/router';
-import { Params } from '@angular/router';
-import {ActivatedRoute} from '@angular/router'
+import {ActivatedRoute, ParamMap} from '@angular/router'
 import { Location }  from '@angular/common';
+import { AddComponent } from './add.component';
+
 import * as _ from 'lodash';
+import 'rxjs/add/operator/switchMap';
+
 @Component({
-     selector: 'edit-component',
-    templateUrl: './edit.component.html',
-    styleUrls: ['../dashboard.component.css'],
-  })
+  selector: 'edit-component',
+  template: `<div *ngIf="todo">
```

Comment:
use template url. not template directly in code

Reply:
ok sir